### PR TITLE
✨ Destinations bigquery+snowflake: Release DV2

### LINKED ISSUE #29728: Real V2 releases for BigQuery and Snowflake
* Remove used "if" branches in code
* remove unused V1 code
* Remove dbt from container
* Update connectors pipeline build code (airbyte-ci/connectors/pipelines/pipelines/actions/environments.py)

### PR DESCRIPTION
**Pull request by edgao**
⚠️ 💀 DO NOT MERGE UNTIL WE ARE CLEARED FOR TAKEOFF 💀 ⚠️ 

closes https://github.com/airbytehq/airbyte/issues/29728. Markdown/yaml changes largely taken from https://github.com/airbytehq/airbyte/pull/29683.

Reviewers: Read each commit in sequence. The first two commits (https://github.com/airbytehq/airbyte/pull/29783/commits/d9aa4d1d96e977defc61cf1a3dca5ec7db43832a + https://github.com/airbytehq/airbyte/pull/29783/commits/4d107681a94b0b2e530d877c763253d87c0bcce4) are basically just `git checkout evan/bigquery-snowflake-v2 -- <stuff>`; the rest are more interesting.

presumably, if anything in here is broken, CI will fail. Not running any interesting tests locally. I did verify that the docker images no longer contain normalization, though.

### CODE REVIEW

**Comment by evantahler** [comment]
Pre-release: https://github.com/airbytehq/airbyte/actions/runs/5957951679

**Review by evantahler — APPROVED** [review]
It's happening 🥳 

👍 metadata changes
👍 docs changes
👍 Dockerfile changes
👍 migration guides
👍 changelogs
👍 spec change

Do we not need a change in `TypingAndDedupingFlag.java`? I guess we still want this for redshift at the moment.

**Comment by edgao** [comment]
> change in TypingAndDedupingFlag

yeah, I want to leave that untouched. The really awkward thing is that I've discovered that we're using it in some files that other connectors depend on (e.g. destination-gcs) - which means naively flipping it to return true would break a lot of things.

there's a bunch of failing tests too. Some are legit (most of the stuff in DestinationAcceptanceTest needs to be turned off, since we're not generating the same raw/normalized tables anymore), some are probably me forgetting to fix something (e.g. connectors base build is failing in CSV-related unit tests, which seems wrong).

**Comment by edgao** [comment]
flipping back to draft while I work through this stuff

**Comment by edgao** [comment]
bigquery standard inserts is broken at this commit, it's writing an empty raw table for some reason. 0c8248d8c75ad2713fa4aefe8efe260f87ad717b is the broken commit.

**Review by Hesperide — CHANGES_REQUESTED** [review]
We can work on updated copy here: https://docs.google.com/document/d/1p9PnS4LOmzSGZsNP2b-WUdoH6HjXBO4uPwEXMOtgsUQ/edit

**Comment by edgao** [comment]
toggling off draft mode to trigger CI again. if this doesn't pass I'm going to just disable the old DATs for now and file an issue to make them work; IMO that shouldn't block release.

File: docs/integrations/destinations/bigquery-migrations.md
Comment (Hesperide) [inline docs/integrations/destinations/bigquery-migrations.md:1]:
```suggestion
# BigQuery Upgrade Guide
```

File: docs/integrations/destinations/snowflake-migrations.md
Comment (Hesperide) [inline docs/integrations/destinations/snowflake-migrations.md:1]:
```suggestion
# Snowflake Upgrade Guides
```

File: docs/integrations/destinations/bigquery-migrations.md
Comment (evantahler) [reply docs/integrations/destinations/bigquery-migrations.md:1]: This is a bigger change as this header if a metadata validation.  This change will need to be in a separate PR.

Comment (Hesperide) [reply docs/integrations/destinations/bigquery-migrations.md:1]: okay, sounds good. thanks for the context

**Review by Hesperide — COMMENTED** [review]
lgtm pending approvals in the breaking change doc.

**Review by Hesperide — APPROVED** [review]

**Comment by evantahler** [comment]
Pre-release: https://github.com/airbytehq/airbyte/actions/runs/5969895524
* ❄️ ---
* 🥞 ---

**Comment by evantahler** [comment]
Pre-release: https://github.com/airbytehq/airbyte/actions/runs/5969895524

* ❄️ XXX
* 🥞 XXX

File: airbyte-integrations/bases/base-java-s3/src/main/java/io/airbyte/integrations/destination/s3/csv/StagingDatabaseCsvSheetGenerator.java
Comment (jbfbell) [inline airbyte-integrations/bases/base-java-s3/src/main/java/io/airbyte/integrations/destination/s3/csv/StagingDatabaseCsvSheetGenerator.java:36]: true?

File: airbyte-integrations/connectors/destination-bigquery/src/main/java/io/airbyte/integrations/destination/bigquery/BigQueryStagingConsumerFactory.java
Comment (jbfbell) [inline airbyte-integrations/connectors/destination-bigquery/src/main/java/io/airbyte/integrations/destination/bigquery/BigQueryStagingConsumerFactory.java:152]: nit and paranoia:
nit: `// In Destinations V2, we will always use the 'airbyte_internal' schema/namespace for raw tables or an override if provided`
paranoia: We are using the override somewhere?

**Review by jbfbell — APPROVED** [review]

File: airbyte-integrations/bases/base-java-s3/src/main/java/io/airbyte/integrations/destination/s3/avro/AvroRecordFactory.java
Comment (edgao) [reply airbyte-integrations/bases/base-java-s3/src/main/java/io/airbyte/integrations/destination/s3/avro/AvroRecordFactory.java:36]: this is intentional - this file isn't actually used by dv2, it's only used by real file destinations (s3/gcs). https://github.com/airbytehq/airbyte/commit/e160ad63e41456d26bd28f208a4da3359d77e44f

File: airbyte-integrations/bases/base-java-s3/src/main/java/io/airbyte/integrations/destination/s3/csv/StagingDatabaseCsvSheetGenerator.java
Comment (edgao) [reply airbyte-integrations/bases/base-java-s3/src/main/java/io/airbyte/integrations/destination/s3/csv/StagingDatabaseCsvSheetGenerator.java:36]: this is to maintain v1 behavior in old destinations (i.e. redshift I think?). V2 destinations are all using the new constructor and explicitly passing true.

File: airbyte-integrations/connectors/destination-bigquery/src/main/java/io/airbyte/integrations/destination/bigquery/BigQueryStagingConsumerFactory.java
Comment (edgao) [reply airbyte-integrations/connectors/destination-bigquery/src/main/java/io/airbyte/integrations/destination/bigquery/BigQueryStagingConsumerFactory.java:152]: ... probably not. this seems like an actual bug.

Comment (edgao) [reply airbyte-integrations/connectors/destination-bigquery/src/main/java/io/airbyte/integrations/destination/bigquery/BigQueryStagingConsumerFactory.java:152]: we ran some manual tests and this seems to work as expected. https://airbytehq-team.slack.com/archives/C03C4AVJWG4/p1693333526355829

**Comment by octavia-approvington** [comment]
Lets do it! <br/> giddddy-up

**Review by octavia-approvington — APPROVED** [review]

**Pull request merged**
